core/math/math_funcs.h

```diff
@@ -438,13 +443,27 @@ class Math {
 		float s = CLAMP((p_s - p_from) / (p_to - p_from), 0.0f, 1.0f);
 		return s * s * (3.0f - 2.0f * s);
 	}
+
 	static _ALWAYS_INLINE_ double move_toward(double p_from, double p_to, double p_delta) {
 		return abs(p_to - p_from) <= p_delta ? p_to : p_from + SIGN(p_to - p_from) * p_delta;
 	}
 	static _ALWAYS_INLINE_ float move_toward(float p_from, float p_to, float p_delta) {
 		return abs(p_to - p_from) <= p_delta ? p_to : p_from + SIGN(p_to - p_from) * p_delta;
 	}
 
+	static _ALWAYS_INLINE_ double rotate_toward(double p_from, double p_to, double p_delta) {
+		double difference = Math::angle_difference(p_from, p_to);
+		double abs_difference = Math::abs(difference);
+		// When `p_delta < 0` move no further than to PI radians away from `p_to` (as PI is the max possible angle distance).
+		return p_from + CLAMP(p_delta, abs_difference - Math_PI, abs_difference) * SIGN(difference);
+	}
+	static _ALWAYS_INLINE_ float rotate_toward(float p_from, float p_to, float p_delta) {
+		float difference = Math::angle_difference(p_from, p_to);
+		float abs_difference = Math::abs(difference);
+		// When `p_delta < 0` move no further than to PI radians away from `p_to` (as PI is the max possible angle distance).
+		return p_from + CLAMP(p_delta, abs_difference - (float)Math_PI, abs_difference) * SIGN(difference);
+	}
```

Comment:
> The first two tests under `// Rotate away.` will most likely fail, as i discovered an issue that if you try moving away from the exact same `from` and `to`, the value will be locked at `from`, and i couldn't find a way to fix this reliably. @kleonc since you wrote the implementation that's currently used, can you try helping with this?

@ettiSurreal Indeed for `p_from == p_to` the `SIGN(difference)` would be zero and hence `p_from` would be returned (even when `p_delta < 0`). This should allow moving/rotating away in such case:
```suggestion
	static _ALWAYS_INLINE_ double rotate_toward(double p_from, double p_to, double p_delta) {
		double difference = Math::angle_difference(p_from, p_to);
		double abs_difference = Math::abs(difference);
		// When `p_delta < 0` move no further than to PI radians away from `p_to` (as PI is the max possible angle distance).
		return p_from + CLAMP(p_delta, abs_difference - Math_PI, abs_difference) * (difference >= 0.0 ? 1.0 : -1.0);
	}
	static _ALWAYS_INLINE_ float rotate_toward(float p_from, float p_to, float p_delta) {
		float difference = Math::angle_difference(p_from, p_to);
		float abs_difference = Math::abs(difference);
		// When `p_delta < 0` move no further than to PI radians away from `p_to` (as PI is the max possible angle distance).
		return p_from + CLAMP(p_delta, abs_difference - (float)Math_PI, abs_difference) * (difference >= 0.0f ? 1.0f : -1.0f);
	}
```
Note that when `p_from == p_to` moving away could be done clockwise/counterclockwise and hence the direction is picked arbitrarily (it could be either `(difference >= 0.0 ? 1.0 : -1.0)` or `(difference > 0.0 ? 1.0 : -1.0)`).